src/stdlib/parse_proto.rs

```diff
@@ -0,0 +1,281 @@
+use super::util::get_message_descriptor;
+use crate::compiler::prelude::*;
+use prost_reflect::ReflectMessage;
+use prost_reflect::{DynamicMessage, MessageDescriptor};
+use std::ffi::OsString;
+use std::path::Path;
+use std::path::PathBuf;
+
+fn proto_to_value(
```

Comment:
Any ideas on reducing code duplication here? Could we pull the shared bit into VRL? We want changes to proto codecs to be in sync with these new functions.

Reply:
My first instinct is to provide the following three functions:
* `fn get_message_descriptor(descriptor_set_path: &Path, message_type: &str,`
* `fn encode_proto(descriptor: &MessageDescriptor, value: Value)`
* `fn parse_proto(descriptor: &MessageDescriptor, value: Value)`

as public functions in src/protobuf/ and expose that through a public module as `vrl::protobuf`. Then we can just use them in the vector crate.
    
Let me know if that makes sense and I can do the changes. Open to other suggestions since I am not super familiar with the code yet.

Reply:
This plan sounds reasonable to me. This will require some effort:

- add the protobuf utils to VRL
- wait for a VRL release
- bump VRL version in Vector
- finally replace duplicate Vector code with `vrl::protobuf` utils

Reply:
This is also a good opportunity to cleanup the `tests/data` a bit. I am not sure if all of those separate files are required. Worth taking a look if some of them can be merged.

Reply:
Thanks, that should be ok. Will try to get a new version of the PR soon.

Reply:
Thank you @flavioc for refactoring, this is starting to look great!

As a next step, just to be sure that the exposed utils are good to go, can you follow up with a draft PR in Vector where:

* Change the VRL version to point to this commit `vrl = { git = "https://github.com/vectordotdev/vrl", rev = "9b25e9954c794f54a6e0e1c8706d79718b72983c" ...` 
* Delete duplicate code and use the utils you exposed in this PR

Reply:
Thanks! So, this is how the vector side will look like: https://github.com/vectordotdev/vector/commit/8538929e7ba7664842430d10d3e6a9ba12c0c54a

Reply:
Is commit that part of a PR? Interested to see the CI checks passing.

Reply:
Yes, I created a draft PR: https://github.com/vectordotdev/vector/pull/20074

Reply:
👍 if those checks I think we can go ahead with this PR